lib/Message.php

```diff
@@ -312,4 +320,23 @@ function getHttpVersion() {
         return $this->httpVersion;
 
     }
+
+    /**
+     * Runs given callback and captures data sent to php://output stream.
+     *
+     * @param callable $callback
+     * @throws \RuntimeException when ob_start() fails to start output buffer
+     * @return string
+     */
+    private function captureCallbackOutput($callback)
+    {
+        $success = ob_start();
+        if ($success === false) {
+            throw new \RuntimeException('Cannot start output buffering');
+        }
+        $callback();
+        $content = ob_get_contents();
```

Comment:
yay, very good point. i don't use this `ob_*` stuff too much :) updated

(note that i can always rebase & squash the commits if needed)